lib/plugins/logviewer/admin.php

```diff
@@ -150,5 +153,61 @@ protected function getFacilities()
         return $facilities;
     }
 
+    private function readEndOfLogFile($logfile, $numberOfLinesLimit)
+    {
+        $lines = [];
+        $fp = fopen($logfile, 'r');
+        
+        fseek($fp, -1, SEEK_END);
+        if(fgetc($fp) == PHP_EOL) { // if the last line is EOL ignore it
+            $pos = -2;
+        } else { // otherwise start from the last character
+            $pos = -1;
+        }
+
+        $currentLine = '';
+
+        while (count($lines) < $numberOfLinesLimit && (-1 !== fseek($fp, $pos, SEEK_END))) {
+            $char = fgetc($fp);
+            if (PHP_EOL == $char) {
+                $lines[] = $currentLine;
+                $currentLine = '';
+            } else {
+                $currentLine = $char . $currentLine;
+            }
+            $pos--;
+        }
```

Comment:
Reading the logfile bytewise seems totally overkill for what we're trying to do here. Seeking to EOF-5MB then reading in chunks (or even a single 5MB chunk) would be much more effective.